File: app/src/main/java/com/example/david/lists/common/buildlogic/NetworkInfoModule.java
package com.example.david.lists.common.buildlogic;

import android.app.Application;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import javax.annotation.Nullable;

import dagger.Module;
import dagger.Provides;

import static android.content.Context.CONNECTIVITY_SERVICE;

@Module
class NetworkInfoModule {
    @Nullable
    @Provides
    NetworkInfo networkInfo(Application application) {
        return ((ConnectivityManager) application
                .getSystemService(CONNECTIVITY_SERVICE))
                .getActiveNetworkInfo();
    }
}

File: app/src/main/java/com/example/david/lists/common/buildlogic/NetworkInfoModule.kt
package com.example.david.lists.common.buildlogic

import android.app.Application
import android.net.ConnectivityManager
import android.net.NetworkInfo
import androidx.core.content.getSystemService
import dagger.Module
import dagger.Provides

@Module
class NetworkInfoModule {
    @Provides
    fun networkInfo(application: Application): NetworkInfo? {
        return application.getSystemService<ConnectivityManager>()
                ?.activeNetworkInfo
    }
}
